source/MaterialXCore/Document.cpp

```diff
@@ -157,12 +157,18 @@ void Document::initialize()
     setVersionString(DOCUMENT_VERSION_STRING);
 }
 
-void Document::importLibrary(ConstDocumentPtr library)
+void Document::importLibrary(ConstDocumentPtr library, const ReadOptions* readOptions)
 {
+    bool skipDuplicatesElements = readOptions && readOptions->skipDuplicateElements;
     for (ElementPtr child : library->getChildren())
     {
-        ElementPtr childCopy = addChildOfCategory(child->getCategory(), child->getName());
-        childCopy->copyContentFrom(child);
+        std::string childName = child->getName();
+        if (skipDuplicatesElements && getChild(childName))
+        {
+            continue;
+        }
+        ElementPtr childCopy = addChildOfCategory(child->getCategory(), childName);
+        childCopy->copyContentFrom(child, skipDuplicatesElements);
```

Comment:
Taking this one step further, it probably makes sense to pass the ReadOptions pointer directly into copyContentFrom(), rather than extracting an intermediate "skipDuplicateElements" boolean.  The full test could then be performed within the "if" statement as:

if (readOptions && readOptions->skipDuplicateElements && getChild(childName))
{
    ...
}

Reply:
I think I'd like to leave it as separate for now as at the highest level we are dealing with reading content into a document and don't think it should be over-generalize to include Element level copy options. Seems for now, some interfaces use the sourceURI option and only this one requires the addition of a skip option.